### DISCUSSION TITLE
Unable to run basic Milvus Features

### DISCUSSION THREAD

**Discussion by harshakhmk** [milvus-io/milvus#29060] [Q&A and General discussion]
I am exploring Milvus and its features, it looked really great, I was going through this [tutorial](https://github.com/towhee-io/examples/blob/main/nlp/text_search/search_article_in_medium.ipynb) and trying it to run but looks like the execution got stuck at this line
```
collection.create_index(field_name= 'title_vector',index_params=index_params)
```
The program got stuck at this line and not proceeding ahead

I am running Milvus as a standalone using docker (guide)[https://milvus.io/docs/install_standalone-docker.md] and using pymilvus==2.2.11

Other way J tried downgrading the pymilvus to 2.2.10 then I was getting below error
Internal Register, Milvus exception (code=1, message= this version of SDK is incompatible with server please down grade you are STK or upgrade your server

**Comment by yhmo** [comment]
The https://milvus.io/docs/install_standalone-docker.md is the latest version v2.3.3, you should install pymilvus 2.3.x
This page tells install the latest pymilvus v2.3.3: https://milvus.io/docs/install-pymilvus.md

**Reply by harshakhmk** [reply]
I installed pymilvus v2. 3.3 following the above documentation, however I am getting an error: 
```
RPC error internal register Milvus exception code =  1, message =  this version of SDK is  incompatible with server please downgrade your SDK or upgrade your server time RPC start

**Reply by yhmo** [reply]
Are you sure you have followed the steps in this page? https://milvus.io/docs/install_standalone-docker.md

**Reply by yhmo** [reply]
Two possible reasons:
1. the server is not v2.3.3, use the command to get the version from its log:
`docker logs [milvus container id] | grep "Versions:"`

If the installation is correct, the output contains "Version: v2.3.3"
```
➜  233 docker logs 56f|grep "Version:"
2023/12/08 04:19:58 maxprocs: Leaving GOMAXPROCS=12: CPU quota undefined

    __  _________ _   ____  ______    
   /  |/  /  _/ /| | / / / / / __/    
  / /|_/ // // /_| |/ / /_/ /\ \    
 /_/  /_/___/____/___/\____/___/     

Welcome to use Milvus!
Version:   v2.3.3
Built:     Fri Nov 10 10:30:54 UTC 2023
GitCommit: f2e972ab
GoVersion: go version go1.20.7 linux/amd64

```

2. the pymilvus is not v2.3.3, use the command to check pymilvus version
`pip3 show pymilvus`

If the pymilvus is installed correctly, the output contains "Version: 2.3.3":
```
Name: pymilvus
Version: 2.3.3
Summary: Python Sdk for Milvus
Home-page: 
Author: 
```




And, double-check your python environment, make sure the script is running in the correct virtual environment.

**Comment by harshakhmk** [comment]
I have followed the steps to install latest Milvus 2.3.3 and make sure, it is correct version of pymilvus
Still I am getting requests.exceptions.ssl error Maxretryerror HTTPSconnectionpole( host =hugging face.co, port= 443 Max retries exceeded with URL: /facebook/dpr-ctx_encoder_single-nq-base/resolve/main/tokenizer_config.jsob caused by SSLError(SSLCertVerificationError,  CERTIFICATE _VERIFICATIN_FAILED: EE certificate key too weak 

Can you help me solve this SSL issue, tried multiple solutions from internet, nothing worked out

**Reply by yhmo** [reply]
HTTPSconnectionpole( host =hugging face.co, port= 443 Max retries exceeded

This is not milvus error. Your network could not access hugging face site.

**Reply by harshakhmk** [reply]
So how found an alternative and solved this issue by following the steps mentioned here https://stackoverflow.com/questions/71692354/facing-ssl-error-with-huggingface-pretrained-models

Would like to know your perspective to resolve this problem, if you have any suggestions

**Reply by yhmo** [reply]
You need a proxy that can access hugging face sites.

Not sure it works for you. Specify the ip in the /etc/hosts (ubuntu os):
```
13.249.205.58 huggingface.co
13.249.205.58 cdn-lfs.huggingface.co
```
